services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtAsyncHttpClient.java

```diff
@@ -87,15 +89,21 @@ public CompletableFuture<Void> execute(AsyncExecuteRequest asyncRequest) {
             new S3CrtResponseHandlerAdapter(executeFuture, asyncRequest.responseHandler());
 
         S3MetaRequestOptions.MetaRequestType requestType = requestType(asyncRequest);
+        String resumeToken = asyncRequest.httpExecutionAttributes().getAttribute(CRT_PAUSE_RESUME_TOKEN);
```

Comment:
what happen if we try to do following
1. clientOne.uploadFile_One
2. clientOne.Pause_one
3. clientOne.uploadFile_Two
4. Wait for uploadFile_Two to complete
5. Then Resume  file One that is  clientOne.Pause_one

Do we need a test case for this ?

Reply:
Do you mean uploading the file, pausing and then uploading the same file from start again? hmm, good question, let me try it. I think the parts uploaded in the first attempt will be overridden by the second attempt.

Reply:
I ran tests with the following order:
1. uploading the file
2. pausing
3. uploading the same file from start 
4. resuming from step2

Step 1, 2 and 3 succeeded. Step 4 threw `NoSuchUploadException`, which makes sense because the object was uploaded successfully in step 3

I'm not sure if we should add a test case for this because it is more of a service behavior